### PR TITLE
Implement Dropdown Widgets.

### LINKED ISSUE #8259: Static drop-downs
Some method's arguments has tag values defined in their documentation. These values should be used in drop down for a given argument. See the GUI1 implementation for details.

### PR DESCRIPTION
**Pull request by MichaelMauderer**
### Pull Request Description

Closes #8259 

[Peek 2023-11-28 12-22.webm](https://github.com/enso-org/enso/assets/1428930/8696bb0f-15de-46a6-9664-a30da931d6df)

### Important Notes

Some TODOs left for after the AST refactoring:
* Updating the AST with the elected value. 
* Getting the Expr ID from the parent node to determine the correct color. 

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: app/gui2/src/components/GraphEditor/widgets/WidgetSelection.vue [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+      <template v-if="showArgumentValue">
+        <span class="name">{{ argName }}</span
+        ><span> {{ selectedValue }} </span>
+      </template>
+      <template v-else>{{ argName }}</template>
+    </span>
+    <div class="WidgetDrodown">
+      <DropdownWidget
+        v-if="showDropdownWidget"
+        :color="parentColor"
```
Comment (Frizi) [inline app/gui2/src/components/GraphEditor/widgets/WidgetSelection.vue]: I think this could just be a constant `color="var(--node-color-primary)"`

File: app/gui2/src/components/GraphEditor/widgets/WidgetSelection.vue [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        :values="tagLabels ?? []"
```
Comment (Frizi) [inline app/gui2/src/components/GraphEditor/widgets/WidgetSelection.vue]: This `??` looks unnecessary, I think `tagLabels` should always be at least an empty array.

File: app/gui2/src/components/GraphEditor/widgets/WidgetSelection.vue [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+<template>
+  <div>
+    <span class="WidgetArgumentName" @pointerdown="showDropdownWidget = !showDropdownWidget">
+      <template v-if="showArgumentValue">
+        <span class="name">{{ argName }}</span
+        ><span> {{ selectedValue }} </span>
+      </template>
+      <template v-else>{{ argName }}</template>
+    </span>
```
Comment (Frizi) [inline app/gui2/src/components/GraphEditor/widgets/WidgetSelection.vue]: this part of the template is incorrect, you shouldn't be handling argument names within the drodpowns explicitly. We still want to show more complicated widget trees on elements that have dropdown choices, e.g to show nested function calls or array widgets within the selected expression. To properly render the dropdown as a wrapper widget around other widgets, you should use `<NodeWidget :input="props.input" />` in some place. That way you allow other matching widgets to attach to the same node.

File: app/gui2/src/components/GraphEditor/widgets/WidgetSelection.vue [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    <div class="WidgetDrodown">
```
Comment (Frizi) [inline app/gui2/src/components/GraphEditor/widgets/WidgetSelection.vue]: Inconsistent class. It should be the same as the component name.  Also, how about naming it `WidgetSingleChoice`?

File: app/gui2/src/components/GraphEditor/widgets/WidgetSelection.vue [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+  if (selectedIndex.value == null) return props.input.info?.defaultValue ?? ''
+  return tagLabels.value[selectedIndex.value]
+})
+const showDropdownWidget = ref(false)
+const showArgumentValue = ref(true)
+
+// When the selected index changes, we update the expression content.
+watch(selectedIndex, (_index) => {
+  console.warn('TODO: Update the expression content once the AST is updated.')
+  const id = '' // TODO: Get the id of the expression once the AST is updated.
```
Comment (Frizi) [inline app/gui2/src/components/GraphEditor/widgets/WidgetSelection.vue]: To make it work at least in case of existing argument, you could use 
```ts
const id = props.input instanceof ArgumentAst ? props.input.ast.astId : undefined
```

**Review by Frizi — CHANGES_REQUESTED** [review]

File: app/gui2/src/components/GraphEditor/widgets/WidgetSelection.vue (hunk 1 above)
Comment (MichaelMauderer) [reply app/gui2/src/components/GraphEditor/widgets/WidgetSelection.vue]: Actually, it seems that `var(--node-color-primary)` can only be used in CSS. But this got me thinking about a workaround: get the style variable from an HTML element. This seems to work as intended now.

Comment (somebody1234) [reply app/gui2/src/components/GraphEditor/widgets/WidgetSelection.vue]: the color prop *is* passed directly to CSS though, so i believe @Frizi's suggestion *should* theoretically work

File: app/gui2/src/components/GraphEditor/widgets/WidgetArgumentName.vue [hunk 6]
```diff
@@ -34,7 +34,7 @@ export const widgetDefinition = defineWidget([ArgumentPlaceholder, ArgumentAst],
 <template>
   <span class="WidgetArgumentName" :class="{ placeholder, primary }">
     <template v-if="showArgumentValue">
-      <span class="name">{{ props.input.info!.name }}</span
+      <span class="value">{{ props.input.info!.name }}</span
```
Comment (Frizi) [inline app/gui2/src/components/GraphEditor/widgets/WidgetArgumentName.vue:37]: Not sure why is this changed? That span is actually representing the argument name, not its value. The "value" is represented by the `NodeWidget` below.

File: app/gui2/src/components/GraphEditor/widgets/WidgetSelection.vue [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+    if (tags == null) return Score.Mismatch
+    return Score.Perfect
+  },
+})
+</script>
+
+<template>
+  <div ref="rootElement" class="WidgetRoot">
+    <span class="WidgetArgumentName" @pointerdown="showDropdownWidget = !showDropdownWidget">
+      <template v-if="showArgumentValue">
```
Comment (Frizi) [inline app/gui2/src/components/GraphEditor/widgets/WidgetSelection.vue:69]: This ref is always `true`, right? I think you could remove this branch completely.

File: app/gui2/src/components/GraphEditor/widgets/WidgetSelection.vue [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+        <NodeWidget :input="props.input" /><span class="value"> {{ selectedValue }} </span>
```
Comment (Frizi) [inline app/gui2/src/components/GraphEditor/widgets/WidgetSelection.vue:70]: I'm not super sure why you need to display the `selectedValue` here, since the `NodeWidget` should on its own represent that value properly. I believe this may lead to repeated text.

**Review by Frizi — APPROVED** [review]
Still have a few questions/nits, but it's in good enough shape to merge into develop and keep working on it there.
